test-suite/generated-src/cpp/color.hpp

```diff
@@ -21,6 +21,9 @@ enum class color : int {
     INDIGO,
     VIOLET,
 };
+constexpr color operator++(color const& r, int increment) noexcept {
```

Comment:
Why are you adding increment, but not decrement or any other arithmatic operations?  Feels like a partial feature.

Reply:
If the increment operator is being used as a way to iterate over a set of enum values, it may be more C++ idiomatic to define an `enum_iterator` class of some kind that follows a general set of rules to reach all values. Otherwise, `operator++` would be sequencing through all _combinations_ of valid flags when used on bit-flag enums, or possibly leading to illegal values (as previously mentioned by @artwyman)